modules/permissions/src/main/java/ca/sickkids/ccm/lfs/permissions/internal/SubjectRestrictionPattern.java

```diff
@@ -77,4 +88,37 @@ public boolean matches()
         // This is not a repository-wide restriction, it only applies to specific nodes
         return false;
     }
+
+    /**
+     * Iteratively search a subject's reference for a match to the {@code targetSubject}.
+     * @param uuid the first subject to check the reference of
+     * @return true if the target was in the chain of references, false if an error occurs or target is not found
+     */
+    private boolean matchesReference(String uuid)
+    {
+        if (StringUtils.equals(uuid, this.targetSubject)) {
+            return true;
+        }
+
+        if (this.session == null) {
+            LOGGER.warn("Could not match subject UUID {}: session not found.", uuid);
+            return false;
+        }
+
+        String nextUuid = uuid;
+        try {
+            Node subject = this.session.getNodeByIdentifier(nextUuid);
+            while (subject.hasProperty("parents")) {
+                nextUuid = subject.getProperty("parents").getString();
+                if (StringUtils.equals(nextUuid, this.targetSubject)) {
+                    return true;
+                }
+                subject = this.session.getNodeByIdentifier(nextUuid);
+            }
+        } catch (RepositoryException e) {
+            LOGGER.error("Failed to find subject UUID {}", nextUuid, e);
```

Comment:
This will usually happen when the user does not have access to the subject node, which is an "expected" scenario.

```suggestion
        } catch (ItemNotFoundException e) {
            LOGGER.debug("Subject UUID {} is inaccessible", nextUuid, e);
        } catch (RepositoryException e) {
            LOGGER.error("Failed to find subject UUID {}", nextUuid, e);
```